app/src/main/java/pl/futurecollars/invoicing/db/sql/SqlInvoiceRepository.java

```diff
@@ -0,0 +1,288 @@
+package pl.futurecollars.invoicing.db.sql;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import javax.annotation.PostConstruct;
+import lombok.AllArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import pl.futurecollars.invoicing.db.InvoiceRepository;
+import pl.futurecollars.invoicing.exceptions.DbException;
+import pl.futurecollars.invoicing.model.Address;
+import pl.futurecollars.invoicing.model.Car;
+import pl.futurecollars.invoicing.model.Company;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.model.InvoiceEntry;
+import pl.futurecollars.invoicing.model.Vat;
+
+@AllArgsConstructor
+public class SqlInvoiceRepository implements InvoiceRepository {
+
+    public static final String SELECT_FROM_INVOICE_QUERY = "select "
```

Comment:
In Java 17 you can try to use """text""" for multiline strings. https://www.baeldung.com/java-multiline-string

Reply:
Done.